pkg/controller/account/account_controller.go

```diff
@@ -89,13 +89,18 @@ func newReconciler(mgr manager.Manager) reconcile.Reconciler {
 		scheme:           mgr.GetScheme(),
 		awsClientBuilder: &awsclient.Builder{},
 	}
+
 	return utils.NewReconcilerWithMetrics(reconciler, controllerName)
 }
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
 func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	// Create a new controller
-	c, err := controller.New("account-controller", mgr, controller.Options{Reconciler: r})
+	maxConcurrentReconciles, err := utils.GetControllerMaxReconciles(controllerName)
```

Comment:
I wonder if it's worth factoring this repeating pattern out into a helper. Would at least make it easier to tweak the behavior e.g. when someone says "These should be warning, not error" :P